src/compiler/scanner.ts

```diff
@@ -278,12 +278,20 @@ module ts {
         return result;
     }
 
-    export function getPositionFromLineAndCharacter(lineStarts: number[], line: number, character: number): number {
+    export function getPositionFromLineAndCharacter(sourceFile: SourceFile, line: number, character: number): number {
+        return computePositionFromLineAndCharacter(getLineStarts(sourceFile), line, character);
+    }
+
+    export function computePositionFromLineAndCharacter(lineStarts: number[], line: number, character: number): number {
         Debug.assert(line > 0);
         return lineStarts[line - 1] + character - 1;
     }
 
-    export function getLineAndCharacterOfPosition(lineStarts: number[], position: number) {
+    export function getLineStarts(sourceFile: SourceFile): number[] {
+        return sourceFile.lineMap || (sourceFile.lineMap = computeLineStarts(sourceFile.text));
+    }
+
+    export function computeLineAndCharacterOfPosition(lineStarts: number[], position: number) {

```

Comment:
For anything new functions introduced, don't export them from `scanner.ts`, expose them from `utilities.ts`.

@mhegazy, this relates to the API test we need to reintroduce; Monday?

I'd also annotate the return types, but that's just me.